test/plugins/test_plugins_handlers.py

```diff
@@ -102,19 +102,19 @@ def test_get_label_entrypoint(plugin_tmpfile):
     assert meta['source'] == 'test_plugin = file_mod'
 
 
-def test_entrypoint_plugin_get_version(plugin_tmpfile):
-    # See gh-2593, an entrypoint plugin whose project/package names are not
-    # equal raises an exception other than ValueError
+def test_entrypoint_plugin_get_version_does_not_raise(plugin_tmpfile):
+    # See gh-2593, wherein an entrypoint plugin whose project/package names
+    # are not equal raised an exception that propagated too far
     distrib_dir = os.path.dirname(plugin_tmpfile.strpath)
     sys.path.append(distrib_dir)
 
-    # load the entry point
     try:
         entry_point = importlib.metadata.EntryPoint(
             'test_plugin', 'file_mod', 'sopel.plugins')
         plugin = handlers.EntryPointPlugin(entry_point)
-        plugin.load()
         plugin.module.__package__ = "FAKEFAKEFAKE"
```

Comment:
Trying to access `plugin.module` before calling `plugin.load()` should raise `RuntimeError`:

https://github.com/sopel-irc/sopel/blob/2e8de02325c4842086eec16f4ee532437643bd5d/sopel/plugins/handlers.py#L279-L283

Before loading and setting the module:

https://github.com/sopel-irc/sopel/blob/2e8de02325c4842086eec16f4ee532437643bd5d/sopel/plugins/handlers.py#L610-L611

… the module is initialized to `None` by the superclass:

https://github.com/sopel-irc/sopel/blob/2e8de02325c4842086eec16f4ee532437643bd5d/sopel/plugins/handlers.py#L277

Reply:
D'oh, I got that backwards. At least it's in an easily-reverted commit.